### PR TITLE
Enhancing network_virtualization script

### PR DESCRIPTION
**Pull request by hathyaga**
network_virtualization script is enhaned to
include failover triggered from HMC and
Client initiated failover tests for Network
virtualized device

Signed-off-by: Harsha Thyagaraja <harshkid@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by hathyaga** [comment]
@narasimhan-v  Added HMC Failover and Client initiated failover tests.
Please review

File: io/net/virt-net/network_virtualization.py [hunk 1]
```diff
@@ -206,6 +208,22 @@ def test_backingdevadd(self):
         if post_add - pre_add != 1:
             self.fail("Failed to add backing device")
 
+    def test_failover(self):
```
Comment (narasimhan-v) [inline io/net/virt-net/network_virtualization.py]: `test_hmcfailover()`

File: io/net/virt-net/network_virtualization.py [hunk 2]
```diff
@@ -380,6 +424,61 @@ def ping_check(self):
...
+            self.log.debug(str(cmdfail))
+            self.fail("Command to set %s as Active has failed" % device)
+
+    def get_backing_device_logport(self):
+        '''
+        Get the logical port id of the
+        backing device
+        '''
+        output = self.backing_dev_list()
+        for i in output:
```
Comment (narasimhan-v) [inline io/net/virt-net/network_virtualization.py]: `for backing_dev in self.backing_dev_list():`

File: io/net/virt-net/network_virtualization.py [hunk 3]
```diff
@@ -380,6 +424,61 @@ def ping_check(self):
...
+                logport = i.split(',')[1:][1].split('/')[1]
+        return logport
+
+    def get_device_logport(self):
+        '''
+        Get the logical port id of the Network
+        virtualized device
+        '''
+        output = self.backing_dev_list()
+        for i in output:
```
Comment (narasimhan-v) [inline io/net/virt-net/network_virtualization.py]: `for backing_dev in self.backing_dev_list():`

File: io/net/virt-net/network_virtualization.py [hunk 4]
```diff
@@ -380,6 +424,61 @@ def ping_check(self):
...
+
+    def get_backing_device_logport(self):
+        '''
+        Get the logical port id of the
+        backing device
+        '''
+        output = self.backing_dev_list()
+        for i in output:
+            if i.startswith('%s,' % self.slot_num):
+                logport = i.split(',')[1:][1].split('/')[1]
```
Comment (narasimhan-v) [inline io/net/virt-net/network_virtualization.py]: why `[1:][1]` ?
Can you paste a sample `i` and `logport` that you want ?

File: io/net/virt-net/network_virtualization.py [hunk 5]
```diff
@@ -380,6 +424,61 @@ def ping_check(self):
...
+                device_logport = i.split(',')[0:][1].split('/')[1]
+        return device_logport
+
+    def is_backing_device_active(self):
+        '''
+        TO check the status of the backing device
+        after failover
+        '''
+        output = self.backing_dev_list()
+        for i in output:
```
Comment (narasimhan-v) [inline io/net/virt-net/network_virtualization.py]: `for backing_dev in self.backing_dev_list():`

File: io/net/virt-net/network_virtualization.py [hunk 6]
```diff
@@ -380,6 +424,61 @@ def ping_check(self):
...
+            if i.startswith('%s,' % self.slot_num):
+                val = i.split(',')[1:][1].split('/')[2]
+        return val
```
Comment (narasimhan-v) [inline io/net/virt-net/network_virtualization.py]: Shouldn't this ideally return a `True` or `False` ?

File: io/net/virt-net/network_virtualization.py [hunk 7]
```diff
@@ -380,6 +424,61 @@ def ping_check(self):
...
+    def get_device_logport(self):
+        '''
+        Get the logical port id of the Network
+        virtualized device
+        '''
+        output = self.backing_dev_list()
+        for i in output:
+            if i.startswith('%s,' % self.slot_num):
+                device_logport = i.split(',')[0:][1].split('/')[1]
+        return device_logport
```
Comment (narasimhan-v) [inline io/net/virt-net/network_virtualization.py]: Will it return only one device logport or an array (if there are more devices) ?

File: io/net/virt-net/network_virtualization.py [hunk 8]
```diff
@@ -380,6 +424,61 @@ def ping_check(self):
             return False
         return True
 
+    def trigger_failover(self, device):
+        '''
+        Triggers failover from HMC
+        '''
+        backing_dev_logport_id = self.get_backing_device_logport()
```
Comment (narasimhan-v) [inline io/net/virt-net/network_virtualization.py]: These 2 variables are not really necessary.

**Comment by hathyaga** [comment]
All changes suggested here is incorporated and sent in PR https://github.com/avocado-framework-tests/avocado-misc-tests/pull/870

Closing this PR

**Pull request closed without merging**